tests/functionaltests/Tests/AssetsLibraryTests/AssetsLibraryTests.mm

```diff
@@ -23,85 +24,175 @@
+                         [WSStorageLibrary
+                             getLibraryAsync:WSKnownLibraryIdPictures
+                                     success:^void(WSStorageLibrary* picturesLibrary) {
+                                         [picturesLibrary.saveFolder
+                                             tryGetItemAsync:fileName
+                                                     success:^void(RTObject<WSIStorageItem>* storageItem) {
+                                                         if (storageItem == nil) {
+                                                             [storageFile
+                                                                 copyOverloadDefaultNameAndOptions:picturesLibrary.saveFolder
+                                                                                           success:^void(WSStorageFile* storageFile) {
+                                                                                               dispatch_group_leave(group);
+                                                                                           }
+                                                                                           failure:onFailure];
+                                                         } else {
+                                                             dispatch_group_leave(group);
+                                                         }
+                                                     }
+                                                     failure:onFailure];
+                                     }
+                                     failure:onFailure];
+                     }
+                     failure:onFailure];
 
-    [assetsLibrary assetForURL:relativeURLToVideo
-        resultBlock:^(ALAsset* asset) {
-            if (asset == nil) {
-                success = NO;
-                message = @"FAILED: Invalid Asset.";
-                dispatch_group_leave(group);
-                return;
-            }
+    dispatch_group_wait(group, DISPATCH_TIME_FOREVER);
+    dispatch_release(group);
 
-            ALAssetRepresentation* _assetRepresentation = [asset defaultRepresentation];
-            if (_assetRepresentation == nil) {
-                success = NO;
-                message = @"FAILED: Invalid Asset Representation.";
-                dispatch_group_leave(group);
-                return;
-            }
+    return success;
+}
 
-            // Check this NSURL after assetForURL completes
-            assetRepresentationUrl = [_assetRepresentation url];
+class AssetsLibrary {
+public:
+    BEGIN_TEST_CLASS(AssetsLibrary)
+    TEST_CLASS_PROPERTY(L"UAP:AppXManifest", L"AssetsLibrary.AppxManifest.xml")
+    END_TEST_CLASS()
+
+    TEST_CLASS_SETUP(AssetsLibraryClassSetup) {
+        bool success = AssetsLibraryTestVideoSetup("AssetsLibraryTestVideo.mp4");
+        success &= FunctionalTestSetupUIApplication();
```

Comment:
Assuming we don't actually run the test if TEST_CLASS_SETUP fails, should we use the non-short circuiting &=?